x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/Engine.java

```diff
@@ -91,14 +91,23 @@ public void writeTo(StreamOutput out) throws IOException {
     private static final ConstructingObjectParser<Engine, String> PARSER = new ConstructingObjectParser<>(
         "engine",
         false,
-        (params, engineName) -> {
+        (params, engineResourceName) -> {
+            final String engineName = (String) params[0];
+            // If engine name is provided, check that it matches the resource name. We don't want it to be updatable
+            if (engineName != null) {
```

Comment:
Check the engine name in case it is present, so it matches the resource name

Reply:
engineName can be null, for example on a PUT where the body that does not include the engine name. It is a valid situation.